src/Juvix/Backends/LLVM/Net/EAC.hs

```diff
@@ -0,0 +1,57 @@
+module Juvix.Backends.LLVM.Net.EAC where
+
+-- TODO ∷ abstract all all imports to LLVM
+
+import qualified Juvix.Backends.LLVM.Codegen as Codegen
+import qualified Juvix.Backends.LLVM.Net.EAC.Types as Types
+import Juvix.Library
+import qualified LLVM.AST.Constant as C
+import qualified LLVM.AST.Operand as Operand
+import qualified LLVM.AST.Type as Type
+
+--------------------------------------------------------------------------------
+-- Interaction Net runner
+--------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------
+-- Reduction rules
+--------------------------------------------------------------------------------
+
+-- TODO ∷ Maybe add metadata at some point?
+
+-- These functions work off the nodeType signature not Types.eac
+
+-- mimic rules from the interpreter
```

Comment:
I wonder if we can split these out neatly into constructor/duplicator rules. If not, it's OK, but it'd be nice.

Reply:
the rules from the original interpreter?

Reply:
Not exactly, just sharing common logic where applicable.